README.md

```diff
@@ -32,3 +28,152 @@
+
+### Streams ###
+A stream is an abstract interface implemented by various objects in Node. Streams are pipes that let you easily read data from a source and pipe it to a destination. [http://nodejs.org/api/stream.html](http://nodejs.org/api/stream.html): 
+
+## RESTful API ##
+### HTTP (methods, headers) ###
+- [RFC 2616: HTTP Methods](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html)
+- [RFC 2616: HTTP Header fields](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html)
+- [How to make a simple HTTP request in Node](http://davidwalsh.name/nodejs-http-request "HTTP request")
+- [Node API: HTTP documentation](http://nodejs.org/api/http.html)
+
+
+### REST principles ###
+- [What is REST? Brief introduction video](https://www.youtube.com/watch?v=7YcW25PHnAA)
+- [REST Principles](http://en.wikipedia.org/wiki/Representational_state_transfer#Architectural_constraints)
+## Express ##
+Express is a minimal and flexible Node.js web application framework that provides a robust set of features for web and mobile applications.
+
+### Starting with Express ###
+- [Generating the application](http://expressjs.com/starter/generator.html)
+- [Example application: Hello World!](http://expressjs.com/starter/hello-world.html)
+- [Basic routing using Express](http://expressjs.com/guide/routing.html)
+- [Using middlewares](http://expressjs.com/guide/using-middleware.html)
+- [Migrating from 3.x to 4.x](https://github.com/strongloop/express/wiki/Migrating-from-3.x-to-4.x)
+- [Express Module Repository](https://github.com/strongloop/express)
+
+### Templating ###
+- [Configuring template engines](http://expressjs-book.com/forums/topic/how-to-use-alternative-non-jade-template-engines-with-express/)
+- [Template engine consolidation library](https://www.npmjs.com/package/consolidate)
+
+## MongoDB ##
+MongoDB is an open source document-oriented database that provides high availability, high performance and automatic scaling.
+
+- [Official documentation](http://docs.mongodb.org/manual/)
+- [MongoDB architecture: Data Model, Query Model and Management](http://www.mongodb.com/mongodb-architecture)
+- [Performance best practices for MongoDB](http://info.mongodb.com/rs/mongodb/images/MongoDB-Performance-Best-Practices.pdf)
+- [Tutorial: creating an API with Node and MongoDB](http://adrianmejia.com/blog/2014/10/01/creating-a-restful-api-tutorial-with-nodejs-and-mongodb/)
+
+
+## Debugging and Profiling ##
+- [How to debug using Node inspector](https://vimeo.com/77870960)
+- [Debugging apps using WebStorm](https://www.jetbrains.com/webstorm/help/running-and-debugging-node-js.html)
+- [Profiling NodeJS applications](http://www.willvillanueva.com/the-node-js-profiling-guide-that-hasnt-existed-profiling-node-js-applications-part-1/)
+- [CPU Profiling](http://www.clock.co.uk/blog/easy-cpu-profiling-in-node-js)
+
+## Others ##
```

Comment:
This section presents relevant modules written by the community, which are proven, and useful to write any NodeJS application